### PR TITLE
Add support for device config to SDK

### PR DESCRIPTION
**Pull request by jake-stark**
PresentationEndpoint with get and getOrCreate endpoints as per [the documentation](https://librarian-regionals.smartthingsgdev.com/api/dossier.st.internal.v20170628#operation/getOrGenerateDeviceConfig)

### CODE REVIEW

File: src/endpoint/presentation.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { Endpoint } from '../endpoint'
+import EndpointClient, { EndpointClientConfig, HttpClientParams } from '../endpoint-client'
+
+
+export enum OperatingMode {
+	EASYSETUP = 'easySetup',
+	DEVICECONTROL = 'deviceControl',
```
Comment (rossiam) [inline src/endpoint/presentation.ts]: Let's use underscores in these names (`EASY_SETUP` and `DEVICE_CONTROL`).

File: src/endpoint/presentation.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	component: string
+	capability: string
+	version: number //default 1
+	values?: CapabilityValue[]
+	visibleCondition?: VisibleCondition
+}
+
+export interface PresentationDeviceConfig {
+	mnmn: string //The name of the manufacturer
+	vid: string //The Vendor Id
```
[the remark this answers is not in the document]
Comment (jake-stark) [reply src/endpoint/presentation.ts]: Screen Shot 2020-03-19 at 9 25 06 AM

I pulled them from the docs, so if any updates are made, the librarian side will have to me changed as well.

File: src/endpoint/presentation.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	NOTCONTAIN = 'notContain',
+	STARTWITH = 'startWith',
+	ENDWITH = 'endWith',
+	EQUAL = '=',
+	NOTEQUAL = '!=',
+	GREATERTHAN = '>',
+	GREATERTHANOREQUALTO = '>=',
+	LESSTHAN = '<',
+	LESSTHANOREQUALTO = '<='
+}
```
[the remark this answers is not in the document]
Comment (jake-stark) [reply src/endpoint/presentation.ts]: I've updated the names, however maybe I should wait and just import this class from CapabilityPresentation?

File: src/endpoint/presentation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rossiam) [reply src/endpoint/presentation.ts]: For now let's go with:

mnmn: the namespace of the integration
vid: an identifier unique to a display configuration for a device

**Comment by rossiam** [comment]
Superseded by https://github.com/SmartThingsCommunity/smartthings-core-sdk/pull/27

**Pull request closed without merging**